cloud/docker_client.go

```diff
@@ -93,20 +87,20 @@ func (c *dockerClientImpl) Init(apiVersion string) error {
 //     3. The image must have the same ~/.ssh/authorized_keys file as the host machine
 //        in order to allow users with SSH access to the host machine to have SSH access
 //        to the container.
-func (c *dockerClientImpl) CreateContainer(ctx context.Context, id string, d distro.Distro, s *dockerSettings) error {
-	dockerClient, err := c.generateClient(d)
+func (c *dockerClientImpl) CreateContainer(ctx context.Context, h *host.Host, name string, s *dockerSettings) error {
```

Comment:
Docker's ContainerCreate function takes a name parameter that allows the user to specify a name for the new container. This is distinct from the ID, and will be automatically generated if not supplied. If we do not care about the name of the container (and will set the container's host.Id to the ID of the successfully started container), I think it might be better to remove this name parameter from the Docker provider and Docker client's functions and just pass nil to ContainerCreate.